graph2tac/tfgnn/tasks.py

```diff
@@ -1281,66 +1337,94 @@ def create_inference_model(self, tactic_expand_bound: int, graph_constants: Grap
+            row_lengths=batch_arg_cnt
         )
-        # [tactic_expand_bound*batch, None(args), None(context)]
+        # [batch, None(tactic_expand_bound*args), None(context)]
         global_arguments_logits = self.global_logits(queries=global_hidden_state_sequences, keys=global_embeddings)
+        # [batch*tactic_expand_bound, None(args), None(context)]
+        global_arguments_logits = tf.RaggedTensor.from_row_lengths(
+            values=global_arguments_logits.values,  # [batch-tactic-arg, None(context)]
+            row_lengths=tactic_arg_cnt
+        )
         
-        # [tactic_expand_bound*batch, None(args), None(context)], [batch, None(args), None(context)]
+        # [batch*tactic_expand_bound, None(args), None(context)], [batch*tactic_expand_bound, None(args), None(context)]
         normalized_local_arguments_logits, normalized_global_arguments_logits = self._log_softmax_logits(
             local_arguments_logits=local_arguments_logits,
             global_arguments_logits=global_arguments_logits)
         
         # TODO: Temporary
-        # [tactic_expand_bound*batch, max(args), max(context)]
+        # [batch*tactic_expand_bound, max(args), max(context)]
         normalized_local_arguments_logits = convert_ragged_logits_to_dense(normalized_local_arguments_logits)
         normalized_global_arguments_logits = convert_ragged_logits_to_dense(normalized_global_arguments_logits)
 
-        # [tactic_expand_bound, batch, max(args), max(local_context)]
+        # [batch, tactic_expand_bound, max(args), max(local_context)]
         normalized_local_arguments_logits = self._reshape_inference_logits(logits=normalized_local_arguments_logits,
                                                                            tactic_expand_bound=tactic_expand_bound)
-        # [tactic_expand_bound, batch, max(args), dynamic_global_context]
+        # [batch, tactic_expand_bound, max(args), dynamic_global_context]
         normalized_global_arguments_logits = self._reshape_inference_logits(logits=normalized_global_arguments_logits,
                                                                            tactic_expand_bound=tactic_expand_bound)
 
         return tf.keras.Model(inputs={self.PROOFSTATE_GRAPH: proofstate_graph, self.TACTIC_MASK: tactic_mask},
                               outputs={self.TACTIC: tf.transpose(top_k_indices),
                                        self.TACTIC_LOGITS: tf.transpose(top_k_values),
-                                       self.LOCAL_ARGUMENTS_LOGITS: normalized_local_arguments_logits,
-                                       self.GLOBAL_ARGUMENTS_LOGITS: normalized_global_arguments_logits})
+                                       self.LOCAL_ARGUMENTS_LOGITS: tf.transpose(normalized_local_arguments_logits, perm=[1, 0, 2, 3]),
+                                       self.GLOBAL_ARGUMENTS_LOGITS: tf.transpose(normalized_global_arguments_logits, perm=[1, 0, 2, 3])})
```

Comment:
It is more natural to work with `[batch, tactic, ...]` instead of `[tactic, batch, ...]`, but the latter is how it was in the code so far.  When I finish beam search, I'll switch it to `[batch, tactic,...]` and get rid of these transpose operations.